pydicom/dataset.py

```diff
@@ -804,40 +756,57 @@ def convert_pixel_data(self):
         elif self._pixel_id != id(self.PixelData):
             already_have = False
 
-        if not already_have:
-            last_exception = None
-            successfully_read_pixel_data = False
-            for x in [h for h in pydicom.config.image_handlers
-                      if h and h.supports_transfer_syntax(self)]:
-                try:
-                    pixel_array = x.get_pixeldata(self)
-                    self._pixel_array = self._reshape_pixel_array(pixel_array)
-                    if x.needs_to_convert_to_RGB(self):
-                        self._pixel_array = self._convert_YBR_to_RGB(
-                            self._pixel_array
-                        )
-                    successfully_read_pixel_data = True
-                    break
-                except Exception as e:
-                    logger.debug("Trouble with", exc_info=e)
-                    last_exception = e
-                    continue
-            if not successfully_read_pixel_data:
-                handlers_tried = " ".join(
-                    [str(x) for x in pydicom.config.image_handlers])
-                logger.info("%s did not support this transfer syntax",
-                            handlers_tried)
-                self._pixel_array = None
-                self._pixel_id = None
-                if last_exception:
-                    raise last_exception
-                else:
-                    msg = ("No available image handler could "
-                           "decode this transfer syntax {}".format(
-                               self.file_meta.TransferSyntaxUID.name))
-                    raise NotImplementedError(msg)
-            # is this guaranteed to work if memory is re-used??
-            self._pixel_id = id(self.PixelData)
+        if already_have:
+            return
+
+        # Find all pixel data handlers that support the transfer syntax
+        suitable_handlers = [hh for hh in pydicom.config.image_handlers
+                             if hh and hh.supports_transfer_syntax(self)]
+
+        # No suitable handlers are available
+        if not suitable_handlers:
+            raise NotImplementedError(
+                "Unable to decode pixel data with a transfer syntax UID of "
+                "'{0}' ({1}) as there are no suitable pixel data handlers "
+                "available.".format(self.file_meta.TransferSyntaxUID,
+                                    self.file_meta.TransferSyntaxUID.name)
+            )
+
```

Comment:
I'd suggest in the NotImplemented message that we help the user a little bit more, with something like "See the 'Supported Transfer Syntaxes' table in pydicom documentation to determine libraries to install" (I think an exact link would be in danger of becoming obsolete, but a table/section title should at least be close).

Reply:
Done